### DISCUSSION TITLE
NV14 with ExpressLRS Status error

### DISCUSSION THREAD

**Discussion by zandorsp** [EdgeTX/edgetx#6133] [Q&A]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

When using the NV14 with external ExpressLRS module the status reports constantly errors

### Expected Behavior

The number of error don´t change

### Steps To Reproduce

1 - Connect an external ExpressLRS module
2 - Create/Select a model with CRFS external module
3 - See the status

### Version

2.7.1

### Transmitter

FlySky NV14

### Anything else?

![Dwkg08_07700b6fa70b34929166e075928547da_00-00-00_00-00-05_2](https://user-images.githubusercontent.com/53715440/175440993-302537f4-7e3d-4174-b40b-cf38c97ea92c.gif)

**Comment by zandorsp** [comment]
If a change the baudrate to 115K , for exemple, the error persists

**Comment by pfeerick** [comment]
As in, it keeps incrementing when you switched to 115k, or that the number remains (you may also have noticed the hz dropped value dropped oddly also)? I would not go above 400k on the NV14 as I don't think the inverters on it are fast enough to do anything higher (reliably, that is). 921k seems mostly stable, but still has somewhat intermittent incrementing handset -> module error count so I wouldn't use it myself without a GPS failsafe recovery system. 

You may need to disconnect and reconnect the module after changing the baud rate so that the module re-detects the new baud rate as while the module *is* automatically power cycled, it seems that some ELRS modules have enough capacitance to stay partially powered during the reset cycle.

**Comment by Neukom720** [comment]
NV14 edgetx 2.7.1 - yep i just battled with the same problem! Doesn't matter the baud rate, the errors just continuously go up with the baudrate just changing how quickly the errors increase, 115k-400k-921k-1.87k all work and I can access the elrs LUA script but the errors just start building up faster and faster as you increase the baud rate, at 1.87k the error is going up 100 every second!  3.75M and 5.25M don't even work and looses connection with the elrs module. Also the elrs Lua script reports 0/50 or zero out of whatever refresh rate your using which is strange because you think it would be dropping frames ya know. Just Flew on 400k using elrs Happymodel ES900tx@50hz and flew fine but it was just a long flight no fast moves or around anything. Both my Happymodel 2.4ghz elrs and my Full size Crossfire 2watt module has the same outcome. Any ideas? Its worry some...thanks guys as always for any help you can give us NV14 boyzzzz haha

**Comment by Neukom720** [comment]
@pfeerick tried pulling the module out after changing baud rate, same outcome. As soon as you plug it back in the error count just continues right on going up....ill try to help troubleshoot if you guys have any other ideas

**Comment by pfeerick** [comment]
> Also the elrs Lua script reports 0/50 or zero out of whatever refresh rate your using which is strange because you think it would be dropping frames ya know.

Unfortunately that's not how it works... these are the errors between the handset and the module - which is *usually* because the speed is pushed beyond what the radio hardware can operate at. It's not related to the control link between the module/transmitter and receiver. So you can have 100% module link/spot on hz rate (i.e. the Bad/Good indicator at the top shows `0/%SET-HZ-RATE%`, and errors on the handset side, and as long as it's a slow creep, because of the data rates we are talking about, they would never propagate through anyway. It's just something to be concerned about until the exact cause can be pinned down... i.e. if the inverters on that particular handset are slow, that's all good and fine... at least it won't get any worse... and you have the _possibility_ of replacing them. 

I'm using 2.4 ELRS, but do have some 900Mhz ELRS so will try that to see if there is any difference...

**Comment by Neukom720** [comment]
@pfeerick tried all my modules, all do the errors, doesn't matter what, my happymodel elrs 2.4ghz and 900mhz, my Ghost Lite flashed to elrs using jr bay adapter, my full size 2watt crossfire, my r9m lite pro on elrs using a jr bay adapter, all do the error thing at whatever baud rate you choose, just trying to save you time on trying different modules cause its def radio side. Also tried elrs v3.0.0rc1 and 2.5.8 both same outcome. Haven't used my tx16s in a long time cause I only use it for fixed wing stuff but I'll dust it off and flash edgetx to verify as well.

**Comment by raphaelcoeffic** [comment]
The error counter you see is actually accurate, and most of these are so called "Framing Errors":
<img width="1282" alt="Screenshot 2022-07-05 at 18 19 34" src="https://user-images.githubusercontent.com/1050031/177372730-5463b2b2-0b02-434f-adad-ed83e912e205.png">

I believe this happens because there is no pull-down resistor neither on the external radio side nor on the external module side. Please note that this is "inverted UART", so that IDLE is LOW.

That being said, it should be pretty much harmless here, as the errors are only artefacts and do not disturb real payload.

**Comment by pfeerick** [comment]
Oh, interesting... yeah, that decay on the end would certainly do it, and as you say, is harmless. I'll have to see if that's what I'm getting. Interesting that it doesn't seem to be happening at the lower baud rates as much in mine... maybe I'm not waiting long enough, or this module actually does have a pulldown or some other filtering.

**Comment by zandorsp** [comment]
> The error counter you see is actually accurate, and most of these are so called "Framing Errors": <img alt="Screenshot 2022-07-05 at 18 19 34" width="1282" src="https://user-images.githubusercontent.com/1050031/177372730-5463b2b2-0b02-434f-adad-ed83e912e205.png">
> 
> I believe this happens because there is no pull-down resistor neither on the external radio side nor on the external module side. Please note that this is "inverted UART", so that IDLE is LOW.
> 
> That being said, it should be pretty much harmless here, as the errors are only artefacts and do not disturb real payload.

That's means that I need a pull-down resistor on S.PORT line in the radio or in the module?

**Comment by raphaelcoeffic** [comment]
> That means that I need a pull-down resistor on S.PORT line in the radio or in the module?

Actually it would be better on the MCU side (between MCU and configurable inverter). We're checking the NV14 hardware, as there seems to be some not-connected pads that were meant for a pull-up resistor at the right place. We're trying to sort this out with FlySky at the moment.

**Comment by Neukom720** [comment]
This is all pretty interesting, can't wait to see where this goes, I hope we can fix it and I'm glad it is currently harmless so I don't have to worry about dropping out of the sky, thanks guys

**Comment by zandorsp** [comment]
I don't know what you made in the nightly releases, but it's seen to work now without errors increasing in status.

**Comment by pfeerick** [comment]
Oh sweet! :) Most likely some changes for the EL18 that also effect the NV14 (but in a good way). Thanks for the update! 🥳

**Comment by guimeraalex020** [comment]
Hey, i have the same problem with the nv14, and the updates aren't solving the problem. What version of edgetx and elrs do you had to solve the problem? Thanks
@zandorsp

**Comment by macfly1fr** [comment]
> Hey, i have the same problem with the nv14, and the updates aren't solving the problem. What version of edgetx and elrs do you had to solve the problem? Thanks @zandorsp

I confirm i have the same issue with all edgetx version. It was not a "real problem" until now when i updated my Tx HappyModel ES24tx 2400 with the latest elrs firmware version 3.1.1 -> when i run elrs's lua script V3 it freeze (in fact it is very very sloowww so unusable ) .
It worked fine (always status error) with all previous elrs firmware (ok with elrs 3.0.1 too ).So i had to flash back to elrs 3.0.1 firmware version in order elrs lua script v3 works again... I suppose this issue is really due to status error continuously of the Nirvana with external module in edgetx.

**Comment by guimeraalex020** [comment]
In my case the lua script is working correctly, i don't understand how the lua is communicating correctly but the radio can't without errors

**Comment by macfly1fr** [comment]
> 



> In my case the lua script is working correctly, i don't understand how the lua is communicating correctly but the radio can't without errors

Your elrs lua script v3 works correctly with a TX module updated with latest elrs "3.1.1" firmware ?

**Comment by zandorsp** [comment]
> Hey, i have the same problem with the nv14, and the updates aren't solving the problem. What version of edgetx and elrs do you had to solve the problem? Thanks @zandorsp



I'm now using EdgeTX 2.8.0 with elrs  3.1.1. The status show some errors, but really not like my first post. 
![photo_2022-12-18_13-56-02](https://user-images.githubusercontent.com/53715440/208310237-2865e6d0-59f4-4f47-827d-e8e70d2421ef.jpg)

**Comment by gagarinlg** [comment]
A few errors are normal, they arise when everything is booting up.

**Comment by guimeraalex020** [comment]
I don't have a few, i have a increasing error number

**Comment by gagarinlg** [comment]
@guimeraalex020 the few errors zandorsp has are OK.

A slightly increasing error number does not stop the Lua script from working.
Most people seem to get better results with EdgeTX 2.8.0. Which version do you have on your radio?

**Comment by guimeraalex020** [comment]
I had 2.7.0 but already upgraded to 2.8.0 and still the same, up to 20.000 errors and increasing.

**Comment by rotorman** [comment]
Do the errors disappear or at least lower, when you use lower baudrate? E.g. 400 kBaud?

**Comment by guimeraalex020** [comment]
They speed down, but they never stop

**Comment by macfly1fr** [comment]
I have the same... errors never stop
My only real issue i that in my case i can't use elrs lua script v3 with module having elrs firmware > 3.0.1

**Comment by zandorsp** [comment]
> I have the same... errors never stop My only real issue i that in my case i can't use elrs lua script v3 with module having elrs firmware > 3.0.1

My NV14 radio was the pull-up resistor from s.port line removed. I don't know really if that's what causes your problems.
![IMG_20221223_152412_371](https://user-images.githubusercontent.com/53715440/209391435-048cb255-c6ba-4d2f-a573-d0f6ba6fd9e7.jpg)

Reference: https://docs.google.com/document/d/1SRegkGT0wnFcTbT9YFNwIuluz5aupOivjGYq4cDiHjE/edit

**Comment by Neukom720** [comment]
> They speed down, but they never stop

Yep I still have this problem, acts the same way, accessing the ELRS lua script also gets screwed up in the higher baudrates or doesn't work at all

**Comment by Neukom720** [comment]
Has anyone verified that removing the pull up resistor from the s.Port line fixes this issue? I'm totally down to remove it as long as it has no I'll effects and removes this issue

**Comment by mmaaddzz** [comment]
I can confirm, after removing that resistor lua script working perfect and errors count is about zero all time.

**Comment by Neukom720** [comment]
@mmaaddzz 
Thanks man, I'll do that tomorrow and hopefully report back with my success as well!!

**Comment by guimeraalex020** [comment]
I did it and now its working!!!!! it was only removing the  pull-up resistor

**Comment by rotorman** [comment]
The resistor in question would be the same as R14 in publicly available EL18 schem: https://raw.githubusercontent.com/open-flysky/EL18-_OpenSourceMaterial/main/sch/EL18-V1.0%E5%8E%9F%E7%90%86%E5%9B%BE(sch)%E5%8F%8A%E7%82%B9%E4%BD%8D%E5%9B%BE.pdf (on EL18 this resistor is not populated in factory).

In case on NV14, the S.PORT line is similarly routed, looking at the schematic, I wonder why it helps to desolder it. Nevertheless great to hear that this helps to solve it.
@richardclli do you know if NV14 S.PORT schem is different to EL18 or have an idea why removing this pull-up makes a difference?

**Comment by macfly1fr** [comment]
> > I have the same... errors never stop My only real issue i that in my case i can't use elrs lua script v3 with module having elrs firmware > 3.0.1
> 
> My NV14 radio was the pull-up resistor from s.port line removed. I don't know really if that's what causes your problems. ![IMG_20221223_152412_371](https://user-images.githubusercontent.com/53715440/209391435-048cb255-c6ba-4d2f-a573-d0f6ba6fd9e7.jpg)
> 
> Reference: https://docs.google.com/document/d/1SRegkGT0wnFcTbT9YFNwIuluz5aupOivjGYq4cDiHjE/edit

Thank you for the information.
I removed this resistor and.... I confirm all problems are solved.
Error status is now very stable , i don't have any issues with last elrs firmware and lua script.

Thank you everybody for the help.

**Comment by OG-ALviK** [comment]
So, for the NV14 users who are having these tipes of issues, to fix them just remove the resistor?

**Comment by zandorsp** [comment]
> So, for the NV14 users who are having these tipes of issues, to fix them just remove the resistor?

Yes, that's it.

**Comment by DeadSeriious** [comment]
**Question**

I am trying to "revive" my Nirvana NV-14 here.... I never really ended up using it at all because I pretty shortly after got a TX16S and it's been my daily driver ever since.

My question though, is I tried updating EdgeTX to ditch that awful UFPV fork and get a normal, up-to-date firmware version running, that seems to be all fine and dandy now. However, using my ELRS module has been loaded with problems. The biggest of which being if I'm running 250mw output power, the radio "dies" almost instantly. I read that this thing cannot support more than 100mw output without having an external power supply/battery. Is this accurate today? Or has anyone managed to find a work around to getting it working without relying on an external pack at >100mw...? 

The other issue that I've been seeing pop up here and there while researching for fixes is issues with ELRS and the gimbal polling rate? What exactly is the issue there and is there anything I can do about THAT issue?

And finally..... does anyone have ANY experience in general with using this radio in 2024/2025 using ELRS? If so, I'd like to know what your experience thus far has been? What (many) hurdles you've had to jump through, etc.

Thanks a bunch. <3 It's too bad this radio was a complete shitstorm during it's time. It had so much potential to be awesome and it was just fumbled it seems, by everyone involved in bringing it to fruition. x_x I hope that the community has had some luck in DIY'ing some fixes to this thing over the years since it's release....

**Comment by richardclli** [comment]
> **Question**
> 
> I am trying to "revive" my Nirvana NV-14 here.... I never really ended up using it at all because I pretty shortly after got a TX16S and it's been my daily driver ever since.
> 
> My question though, is I tried updating EdgeTX to ditch that awful UFPV fork and get a normal, up-to-date firmware version running, that seems to be all fine and dandy now. However, using my ELRS module has been loaded with problems. The biggest of which being if I'm running 250mw output power, the radio "dies" almost instantly. I read that this thing cannot support more than 100mw output without having an external power supply/battery. Is this accurate today? Or has anyone managed to find a work around to getting it working without relying on an external pack at >100mw...?
> 
> The other issue that I've been seeing pop up here and there while researching for fixes is issues with ELRS and the gimbal polling rate? What exactly is the issue there and is there anything I can do about THAT issue?
> 
> And finally..... does anyone have ANY experience in general with using this radio in 2024/2025 using ELRS? If so, I'd like to know what your experience thus far has been? What (many) hurdles you've had to jump through, etc.
> 
> Thanks a bunch. <3 It's too bad this radio was a complete shitstorm during it's time. It had so much potential to be awesome and it was just fumbled it seems, by everyone involved in bringing it to fruition. x_x I hope that the community has had some luck in DIY'ing some fixes to this thing over the years since it's release....

NV14 internal power output is not stable enough to drive external ELRS module.  This can cause a blackout in NV14 if the battery voltage is low + ELRS power consumption is high.  If you really love this form factor, consider an EL18 instead.

**Reply by Neukom720** [reply]
Get the el18 mini jr module adapter, it's has a xt30 on the bottom of it and use a 2s battery connected, run any power you want and battery issues are a thing of the past and if you haven't yet, remove the pull up resistor on the s.port line. My nv14 has been my daily since edgetx released! I only run elrs on it usually 915mhz @200hz because it needs even less output power than 2.4ghz but still feels great to fly at 200hz and if you want 100mw at 915mhz can go very far if you want to just not worry about power issues, only 2.4ghz needs more power for range but even so elrs goes insanely far on low output power on either band but obviously 915mhz will go farther on less power so that's my advice to switch to 915mhz and buy the el18 mini jr adapter with xt30 built in on the bottom.

**Reply by DeadSeriious** [reply]
Awesome! So you've got yours working/have a work-around and still use yours today?! Is it on the latest (or close to) edgeTX version with no issues? What about the polling rate issue of the gimbals?

Thanks for responding man! Got a pick of your NV-14 with the module adapter?

**Reply by pfeerick** [reply]
I suspect he means Nano module adapter. The JR Micro adapter is for the larger form factor modules, and does not provide external power. The Nano module adapter has a (edit: typo, 30 not 60!) XT30 connector on the bottom. I've not had any issue using ELRS with NV14, but I also don't use the higher power (i.e. above 100mw) modules with it, and have a NV14 that has had the power circuitry modified. Even so, next option would have been to use a module with external battery support. And then finally move to something like the EL18. 
 
https://www.amazon.com/Flysky-Transmitter-External-Module-Adapter/dp/B0BQC32W15

**Reply by Neukom720** [reply]
Yes I meant nano module adapter and it doesn't have an xt60 on the bottom, it's an xt30 on the bottom. Also if used you must only use a 2s battery connected to the xt30. Mine works with no power issues, used it at 1w on 915mhz elrs at 200hz. Also you can just buy one of the Radiomaster Elrs modules in either 2.4ghz or 915mhz flavors with the xt30 connector built into the modules and they're well built Elrs modules too. Others have external power on them too like Betafpv makes a few and quite a few others. I haven't looked into the polling rates but I thought the polling rate issues was just with the new EL18 and not the older NV14 but I could be wrong on that one. It's my main fpv radio and I use custom finger grips on the back. My wrist is in bad shape after being hit by a car 8 years ago and after a bunch of screws and titanium plates in my wrist and 2 years of physical therapy, there still not great but the best they can be. The only radio I can comfortable use anymore is my NV14 and My Tango 2 but my NV14 is more comfortable for me. I have no issues running later versions of Edgetx, I'd say I updated to the latest Edgetx like a year ago, I don't have the radio in front of me, I'm on a long term job site out of state. I have only done a few small mods inside the radio, I did the s.port pull down resistor removal to rid myself of errors, my Elrs still worked well before that but will errors and my Lua was slower than normal but it still worked. I can't thank the Edgetx team enough, they made flying fpv possible for me again! Also the Elrs team too because I flew crossfire in Forced 150hz mode which has terrible range because I think it's in FLRC mode when in 150hz forced mode and when in 50hz mode it goes to Lora but for me 50hz doesn't feel good to fly. So 915mhz ELRS at 200hz feels great and has killer range! I don't need any more than 200hz so It doesn't make sense for me to run 2.4ghz but I am starting to use twin antenna modes on receivers and transmitter modules to run Gemini Modes because it just adds a lot of killer features and backup. Gemini allows for basically zero packet loss and perfect crisp control even at the edge of range or tough RF conditions! Thanks to everyone who contributed to these amazing open source communities!

**Reply by pfeerick** [reply]
Polling rate for the gimbals applies for all  Flysky hall gimbals, as they don't sync with the RF packet rate. Just meant using higher ELRS packet rates was relatively pointless. Flysky have been working on a firmware update for the gimbals to fix this.

**Reply by jwidess** [reply]
Adding to this, I recently got my NV14 back out after sitting in a box for ~7 years... I also acquired a **Happymodel ES24TX Pro** 1W 2.4G ExpressLRS TX module for the JR bay and soon after ran into the problems discussed here. 
I went ahead and followed all of the HW mods in the fantastic ["Power Mod & Bay Mod of the Flysky Nirvana NV14"](https://docs.google.com/document/d/1SRegkGT0wnFcTbT9YFNwIuluz5aupOivjGYq4cDiHjE) by qba667, Stefan73 et al. After completing all of these mods, my ES24TX Pro 1W works alright, no packet loss in the ELRS Lua script, my voltage sag is much less, the JR Bay turns off properly, and the JR bay now gets the correct 6.5V output, much improved! However, I still can't go above 100mW of transmit power on the ES24TX Pro without the module crashing, due to voltage sag issues, browning out the transmitter* (Refer to note below)
~~That being said, I don't expect to need >250mW of TX power ever, so I'm very happy with how my NV14 is now~~, and I'm very thankful to the authors for figuring everything out!
**EDIT**: I'm beginning to reach the end of my reception in certain scenarios at 100mW of TX power, so I plan to modify the ES24TX Pro 1W module to accept an external 2S battery, which I'll use to power it. Either that, or I'll just upgrade to the Nomad Dual 1W Gemini module.
For anyone who is interested, I have created an "**improved**" version of the HW mod document, which can be found here,
**NV14 Power & Bay Mod - V2:** https://docs.google.com/document/d/1gauT_LnciE42GizPJLEwAzpfjIcEYqL82MlzOYP5KFQ

**EDIT 2:** I've finished my designs for externally powering and mounting the ES24TX Pro module on the rear of the remote. This really has turned my NV14 into a capable transmitter! The [3D models can be found on Printables here](https://www.printables.com/model/1752848-flysky-nv14-nirvana-external-battery-holder-2s-186).

**EDIT 3:** Finally put together my [full Reddit post about my NV14 here](https://www.reddit.com/r/radiocontrol/comments/1u8v0cw/flysky_nv14_nirvana_mods_fixes_and_elrs_in_2026/).

*I have confirmed this with my o-scope, there seems to be an issue with the NV14's boost converter, as even with the increased boost voltage at 6.5V, its not enough and at 250mW or above strange hiccups occur that brown out the module. I've even tried adding upto 1mF of capacitance to the 6.5V VBAT power rail on the module, but its still not enough.

> [!IMPORTANT]
>  Also be aware that there is a bug with EdgeTX >v2.10.6 on the NV14, resulting in a complete loss of all switch functions after some time of the radio being on. I've personally experienced this problem on my remote. I'd recommend sticking with v2.10.6 until this issue is resolved. **EdgeTX Issue:** https://github.com/EdgeTX/edgetx/issues/6112
